### DISCUSSION TITLE
Passing variables into partials

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#4089] [Feature Requests]
### Suggest an idea for ExpressionEngine

Imagine you need to reuse a piece of code (a component) several times in a template. Right now, we have two options:

1. Copy and paste the component code, changing the variables as needed.
2. Use an embedded template multiple times, which impacts performance because embedded templates consume more resources.

Having the capability to pass variables into partials, similar to how we can with embedded templates, would be incredibly helpful.

**Comment by acutedeveloper** [comment]
This can be achieved using the EE Pro Variables Addon. Comes OTB with your EE7 install.

So for example I have a button component. Create a new variable called `card_button`.

You can also manually do this in the theme folder by creating a `_variables` folder and then a `card_button.html` file.

Set your component markup:

```
<button class="btn {cta_color}">
    {cta_text}
</button>
```

Then in your template call the component like so:

```
{exp:pro_variables:single var="card_button"
          preparse:cta_color="btn-primary-red"
          preparse:cta_text="{article_cta_text}"
}
```

More info in the EE Docs

If you want a more versatile component approach, consider looking at Coilpack for EE. This now allows you to use Twig or Blade template engines. Both are very common PHP template engines for a component based approach.

**Reply by shbchk** [reply]
The only problem is that variables cannot process EE tags, they are static chunks of code

**Reply by acutedeveloper** [reply]
No you are not understanding. There is EE variables which are static chunks of code.

Then there is the EE Pro Variables add-on where you can parse variables. All info in the EE docs.

**Reply by shbchk** [reply]
I'll take a closer look, thanks!
